### ISSUE TITLE
Protobuf module started making it impossible to access the UCAR THREDDS server using siphon TDSCatalog

### ISSUE THREAD

**Issue by edrewitz** [Unidata/siphon#752]
Hi all!

For some reason I started getting these errors when trying to access the UCAR THREDDS server using siphon TDS Catalog. I'm not sure why this is happening all of a sudden. I had to uninstall and reinstall Python and miniconda this week as there was an issue with one of the DLLs that began this past weekend for some reason. I am using the same versions of Python and miniconda as before but things aren't working quite right with siphon ever since I had to uninstall and reinstall everything. 

I also tried to downgrade the version of protobuf to 3.20.1 from 3.20.3 and it wouldn't let me. 

This is the error message: 

TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

I copied and pasted the portion of the code that throws me this error. Please let me know if you have any idea for a solution. Thank you so much! 

```python
from datetime import datetime, timedelta
from siphon.catalog import TDSCatalog

# Dates and times
dt = datetime.utcnow()
year = dt.year
month = dt.month
day = dt.day
hour = dt.hour
minute = dt.minute

# Times for METAR reports
if minute <30:
    dt1 = datetime.utcnow() 
if minute >=30:
    dt1 = datetime.utcnow() - timedelta(minutes=minute)
date = datetime(year, month, day, hour, minute)

# Times used for change plots
if minute <50:
    dt2 = datetime.utcnow() 
    yr = dt2.year
    mon = dt2.month
    dy = dt2.day
    hr = dt2.hour
    date1 = datetime(yr, mon, dy, hr) - timedelta(hours=2)
    date2 = date1 - timedelta(hours=24)
if minute >=50:
    dt2 = datetime.utcnow() - timedelta(minutes=minute)
    yr = dt2.year
    mon = dt2.month
    dy = dt2.day
    hr = dt2.hour
    date1 = datetime(yr, mon, dy, hr) - timedelta(hours=1)
    date2 = date1 - timedelta(hours=24)

#-------------------------------------------------------------------------------------------------------
# Data ingest 
#------------------------------------------------------------------------------------------------------

# Data Access for current time
rtma_cat = TDSCatalog('https://thredds.ucar.edu/thredds/catalog/grib/NCEP/RTMA/CONUS_2p5km/RTMA_CONUS_2p5km_'+date1.strftime('%Y%m%d_%H00')+'.grib2/catalog.xml')
rtma_data = rtma_cat.datasets['RTMA_CONUS_2p5km_'+date1.strftime('%Y%m%d_%H00')+'.grib2'].remote_access(use_xarray=True)

# Data Access for 24 hours ago
rtma_cat_24 = TDSCatalog('https://thredds.ucar.edu/thredds/catalog/grib/NCEP/RTMA/CONUS_2p5km/RTMA_CONUS_2p5km_'+date2.strftime('%Y%m%d_%H00')+'.grib2/catalog.xml')
rtma_data_24 = rtma_cat_24.datasets['RTMA_CONUS_2p5km_'+date2.strftime('%Y%m%d_%H00')+'.grib2'].remote_access(use_xarray=True)

# Parses data
rtma_data = rtma_data.metpy.parse_cf()
rtma_data_24 = rtma_data_24.metpy.parse_cf()

```

- Problem description: this should explain **why** the current behavior is a problem and why
  the expected output is a better solution.
- Expected output
- Which platform (Linux, Windows, Mac, etc.)
- Versions. Include the output of:
  - `python --version`
  - `python -c 'import siphon; print(siphon.__version__)'

**Comment by edrewitz** [comment]
Forgot to mention, I am using Windows OS and Python 3.11.4.

**Comment by dopplershift** [comment]
I assume since you closed this you got it figured out?

**Comment by edrewitz** [comment]
> I assume since you closed this you got it figured out?

Good day Ryan, 

Yes I found the fix. I'll speak about it here since it could potentially be helpful to others if they encounter the same issue. 

So as I stated above, at least one of the DLLs for my Python became corrupted and I had to uninstall both python and miniconda3 and reinstall. I was using Python 3.11.4. When I reinstalled Python 3.11.4 I installed miniconda 3.11. One weird thing I encountered when troubleshooting miniconda 3.11 was when I had the path to environment variable box checked, the scripts that utilize the cfgrib module ran fine in the windows task scheduler but the scripts using siphon did not. Likewise, when I uninstalled and re-installed miniconda 3.11 and left the path to environment variable box unchecked, the scripts that utilize cfgrib did not run in the windows task scheduler error code 0x1 but the scripts using siphon ran fine. 

The fix turned out to be using miniconda 3.10 with having the path to environment variable box checked (I had all 4 boxes checked when setting up miniconda 3.10) and that fixed everything as now both the scripts that use cfgrib and siphon both work. It appears that protobuf module that siphon uses isn't friendly with miniconda 3.11 yet but works fine with miniconda 3.10. 

I hope this helps. 

Regards,

Eric Drewitz

**Comment by dopplershift** [comment]
Hrrmm, I sense there's something more complicated going on here. Siphon's test suite (including the protobuf-based functionality) is passing locally for me on both Python 3.11 and 3.12 on arm64 macOS. We need to do some more work here on GitHub to get the CI runs back to get a good test on Windows here to be sure, but I'm concerned this is more environment related.
